test/e2e/integrationcluster/backup_test.go

```diff
@@ -202,9 +203,32 @@ var _ = Describe("Backup", func() {
 			newFullSnap, newDeltaSnapList, err := miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
 			Expect(err).ShouldNot(HaveOccurred())
 
-			Expect(latestSnapshots.FullSnapshot).Should(Or(Equal(fullSnap), Equal(newFullSnap)))
-			for i, snap := range latestSnapshots.DeltaSnapshots {
-				Expect(snap).Should(Or(Equal(deltaSnapList[i]), Equal(newDeltaSnapList[i])))
+			if latestSnapshots.FullSnapshot == nil {
+				Expect(latestSnapshots.FullSnapshot).Should(Or(Equal(fullSnap), Equal(newFullSnap)))
+			} else {
+				// prefix is not determined during http call So prefix can't be tested here.
+				Expect(*latestSnapshots.FullSnapshot).To(MatchFields(IgnoreExtras, Fields{
```

Comment:
I don't think we need to match each field here, since `.Equals()` already does that for us right?

Reply:
`.Equals()` won't work here as I mentioned in the commented line.

Reply:
Please ignore my comment. The [old code](https://github.com/abdasgupta/etcd-backup-restore/blame/13ddcaa7ec25be4b23f204124ac3f718fe9920c9/test/e2e/integrationcluster/backup_test.go#L205-L208) was also checking both old and new snapshots. It was using `Equals` but because of prefixes in the snapshot paths, use of `MatchFields` was required) 👍 

@shreyas-s-rao Any idea why the old code was checking both old and new snapshots?